packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -43,6 +47,71 @@ export function SummaryContainer() {
     [projects]
   );
 
+  const chainIds = useMemo(
+    () => Object.keys(projectsByChain).map(Number),
+    [projectsByChain]
+  );
+
+  /** How much of the voting token for a chain does the address have*/
+  const [tokenBalancesPerChain, setTokenBalancesPerChain] = useState<
+    Map<ChainId, bigint>
+  >(new Map());
+  useEffect(() => {
+    const runner = async () => {
+      const newMap = new Map(tokenBalancesPerChain);
+      await Promise.all(
+        chainIds.map(async (chainId) => {
+          const votingToken = getVotingTokenForChain(chainId);
+          const { value } = await fetchBalance({
+            address: address ?? zeroAddress,
+            token:
+              votingToken.address === zeroAddress
+                ? undefined
+                : votingToken.address,
+            chainId,
+          });
+          newMap.set(chainId, value);
+        })
+      );
+      setTokenBalancesPerChain(newMap);
+    };
+    runner();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [address, chainIds, getVotingTokenForChain]);
```

Comment:
The pattern of combining `useState` and `useEffect` can often be replaced with a `useSWR` hook. That way we get caching (memo), error and loading states.

Reply:
will do in next iteration, don't wanna change too much until we have tests for this page, but very much agree on that.